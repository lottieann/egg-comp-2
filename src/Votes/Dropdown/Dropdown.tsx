import * as React from 'react'

import styled from 'styled-components'

export interface DropdownProps {
    options: number[],
    label: string
}

const StyledDropdown = styled.div`
  width: 200px;
`

const StyledSelect = styled.select`
  width: 150px;
`

export const Dropdown = (props: DropdownProps) => {
  return (
    <StyledDropdown>
        <label>{props.label}</label>
      <StyledSelect defaultValue="hoose your egg">
        {props.options.map((el, index) => {
          return <option value={el} key={index}>{el}</option>
        })}
      </StyledSelect>
    </StyledDropdown>
  )
}
