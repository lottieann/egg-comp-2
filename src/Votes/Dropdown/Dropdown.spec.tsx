import * as React from 'react'

import 'jest-styled-components'

import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import { Dropdown } from './Dropdown'

const mockProps = {
    options: [1, 2, 3, 4, 5, 6, 7, 8, 9]
}

const setup = () => {
  return <Dropdown options={mockProps.options} label="label"/>
}

describe('Dropdown', () => {
  it('should render Dropdown correctly', () => {
    const { container } = render(setup())
    expect(container).toMatchSnapshot()
  })
})