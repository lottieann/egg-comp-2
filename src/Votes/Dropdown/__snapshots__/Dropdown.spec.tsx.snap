// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown should render Dropdown correctly 1`] = `
.c0 {
  width: 200px;
}

.c1 {
  width: 150px;
}

<div>
  <div
    class="c0"
  >
    <label>
      label
    </label>
    <select
      class="c1"
    >
      <option
        value="1"
      >
        1
      </option>
      <option
        value="2"
      >
        2
      </option>
      <option
        value="3"
      >
        3
      </option>
      <option
        value="4"
      >
        4
      </option>
      <option
        value="5"
      >
        5
      </option>
      <option
        value="6"
      >
        6
      </option>
      <option
        value="7"
      >
        7
      </option>
      <option
        value="8"
      >
        8
      </option>
      <option
        value="9"
      >
        9
      </option>
    </select>
  </div>
</div>
`;
